File: LeetCode/541. Reverse String II/Solution.cpp
#include <bits/stdc++.h>
using namespace std; 

class Solution {
    public:
        string reverseStr(string s, int k) {
            int n = s.length();
            for(int i = 0; i < n; i += 2 * k) {
                int L = i;
                int R = min(i + k - 1, n - 1);
                while(L < R) {
                    swap(s[L++], s[R--]);
                }
            }
            return s;
        }
};

File: LeetCode/541. Reverse String II/Solution.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        a = list(s)
        n = len(a)
        for i in range(0, n, 2 * k):
            L, R = i, min(i + k - 1, n - 1)
            while L < R:
                a[L], a[R] = a[R], a[L]
                L += 1
                R -= 1
        return "".join(a)